### PR TITLE
fetch real data from the server

### PR DESCRIPTION
**Pull request by ananayis**
the app now can get data from the server.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -43,7 +43,14 @@ dependencies {
...
     implementation 'androidx.legacy:legacy-support-v4:1.0.0'
+
     testImplementation 'junit:junit:4.+'
     androidTestImplementation 'androidx.test.ext:junit:1.1.3'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+
+    // Retrofit
+    implementation "com.squareup.retrofit2:retrofit:2.9.0"
+    implementation 'com.squareup.retrofit2:converter-gson:2.3.0'
+//    implementation 'com.google.code.gson:gson:2.8.7'
```
Comment (mousamk) [inline app/build.gradle:54]: Please remove this comment.

File: app/src/main/java/com/parinaz/weblog/PostRepository.kt [hunk 2]
```diff
@@ -1,116 +1,153 @@
...
-                title = "Lorem ipsum",
-                body = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore...",
-            ),
+//    fun getPost(): List<Post>{
+//        val posts = listOf(
+//            Post(
+//                id= 1,
+//                title = "Hello World! hello world hello world hello world hello world",
+//                body = "This is my first post, welcome to my blog!",
+//            ),
```
Comment (mousamk) [inline app/src/main/java/com/parinaz/weblog/PostRepository.kt:17]: Please remove these comments.

File: app/src/main/java/com/parinaz/weblog/PostRepository.kt [hunk 3]
```diff
@@ -1,116 +1,153 @@
...
+                val posts = response.body().orEmpty()
+                callBack(posts)
+            }
 
-        )
-        return posts
+            override fun onFailure(call: Call<List<Post>>, t: Throwable) {
+                t.printStackTrace()
+                //TODO("Not yet implemented")
+                //Log.e("GetPosts", "Error happened: ${t.message}")
```
Comment (mousamk) [inline app/src/main/java/com/parinaz/weblog/PostRepository.kt:134]: Please remove these comments.

File: app/src/main/java/com/parinaz/weblog/network/WeblogApiService.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    @GET("/posts/{postId}/comments")
+    fun getComments(@Path("postId") postId: Long) : Call<List<Comment>>
+}
+
+object WeblogApi{
+    val retrofitService : WeblogApiService by lazy {
+        retrofit.create(WeblogApiService::class.java)
+    }
+}
```
Comment (mousamk) [inline app/src/main/java/com/parinaz/weblog/network/WeblogApiService.kt:31]: Please add a new line to the end of this file.

**Review by mousamk — COMMENTED** [review]
This issue is not finished yet. You need to add a spinner (progressbar) and show it while the data are being fetched.

File: app/build.gradle (hunk 1 above)
Comment (ananayis) [reply app/build.gradle]: done

File: app/src/main/java/com/parinaz/weblog/PostRepository.kt (hunk 2 above)
Comment (ananayis) [reply app/src/main/java/com/parinaz/weblog/PostRepository.kt]: done

File: app/src/main/java/com/parinaz/weblog/PostRepository.kt (hunk 3 above)
Comment (ananayis) [reply app/src/main/java/com/parinaz/weblog/PostRepository.kt]: done

File: app/src/main/java/com/parinaz/weblog/network/WeblogApiService.kt (hunk 4 above)
Comment (ananayis) [reply app/src/main/java/com/parinaz/weblog/network/WeblogApiService.kt]: done

**Review by mousamk — APPROVED** [review]

**Pull request merged**